src/control/SConscript

```diff
@@ -43,13 +45,19 @@ def gen_build_id():
     return '0x' + buildid.decode()
 
 
-def go_ldflags():
+def go_ldflags(benv):
     "Create the ldflags option for the Go build."
 
+    build_host = ''
+    if not is_release_build(benv):
+        build_host = socket.getfqdn()
+    build_time = datetime.now(timezone.utc).astimezone().isoformat()
     Import('daos_version', 'conf_dir')
     path = 'github.com/daos-stack/daos/src/control/build'
     return ' '.join([f'-X {path}.DaosVersion={daos_version}',
                      f'-X {path}.ConfigDir={conf_dir}',
+                     f'-X {path}.BuildTime={build_time}',
```

Comment:
This has the unfortunate side effect of causing an expensive re-build of the go components for every source build regardless of if anything has changed.

If there is a go version of the c __date__ or __time__ macros then you should use that, if not there's a way of telling scons that a subset of options shouldn't be compared against the cache when considering rebuilds, if you need me to I can look into this.